homeassistant/helpers/entity_registry.py

```diff
@@ -127,8 +127,8 @@ def async_get_or_create(self, domain, platform, unique_id, *,
         if entity_id:
             return self._async_update_entity(
                 entity_id,
-                config_entry_id=config_entry_id,
-                device_id=device_id,
+                config_entry_id=config_entry_id or _UNDEF,
```

Comment:
I think the better solution is to use `_UNDEF` instead of `None` as the default value.

Reply:
There was some reason why I didn't, I can't really remember why though. Will double check.

Reply:
Did you double check? :)

Reply:
I think what it came down to was that if you made `_UNDEF` the default instead of `None`, then further down when you're calling the `RegistryEntry` constructor you'd have to use something like `None if device_id is _UNDEF else device_id`.

Reply:
I think it's good to go.